platform/core/src/services/ToolBarService/ToolbarService.ts

```diff
@@ -2,35 +2,43 @@ import merge from 'lodash.merge';
 import { CommandsManager } from '../../classes';
 import { ExtensionManager } from '../../extensions';
 import { PubSubService } from '../_shared/pubSubServiceInterface';
-import type { RunCommand, Commands } from '../../types/Command';
+import type { RunCommand, Commands, Command } from '../../types/Command';
 
 const EVENTS = {
   TOOL_BAR_MODIFIED: 'event::toolBarService:toolBarModified',
   TOOL_BAR_STATE_MODIFIED: 'event::toolBarService:toolBarStateModified',
 };
 
-export type ButtonListeners = Record<string, RunCommand>;
-
-export interface ButtonProps {
-  primary?: Button;
-  secondary?: Button;
-  items?: Button[];
+enum ButtonInteractionType {
+  ACTION = 'action',
+  TOGGLE = 'toggle',
+  TOOL = 'tool',
 }
 
-export interface Button extends Commands {
+export interface ButtonProps {
```

Comment:
Can you change this to a type please?  That works better for extending it with customizations and other info.